.github/workflows/ci-publish.yml

```diff
@@ -0,0 +1,127 @@
+# Copyright 2022 JanusGraph Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: CI Release artifacts publish
+
+on:
+  push:
+    tags:
+      - v*.*.*
+
+jobs:
```

Comment:
Wouldn't be cool to split up this thing into multi jobs? So we could run the most steps also on our main branch.

Reply:
I don't fully understand this one. When we trigger this release publish we trigger it for specific commit and not for specific branch.
It doesn't matter if it is for 0.6 branch or for master branch. Only the tag matter (which is assigned to specific commit on a specific branch).
Running this publish for all multiple branches doesn't make sense because we may have 0.6.1 release but not 1.0.0 release yet. 
That said, I maybe just misunderstood the question here

Reply:
We could run all steps beside the release on all main branch. So, we could provide snapshot every day or month. We also see if most of the stuff is working correctly.

Reply:
Agree, we could do that for sure and probably it will be cool to provide snapshot releases for all commits but I think it should be in another PR.
Also, I had some vision about snapshot releases for each commit here: 
https://lists.lfaidata.foundation/g/janusgraph-dev/topic/discussion_release_snapshot/84922650?p=,,,20,0,0,0::recentpostdate/sticky,,,20,2,0,84922650,previd=1642179578049843803,nextid=1613096947766667047&previd=1642179578049843803&nextid=1613096947766667047

I think we could discuss the approach for snapshot releases